src/serializer/ResidualHeapVisitor.js

```diff
@@ -1130,38 +1145,46 @@ export class ResidualHeapVisitor {
         fixpoint_rerun();
       },
       visitModifiedBinding: (modifiedBinding: Binding) => {
-        invariant(additionalFunctionInfo);
-        let { functionValue } = additionalFunctionInfo;
-        invariant(functionValue instanceof ECMAScriptSourceFunctionValue);
-        let residualBinding = this.getBinding(modifiedBinding.environment, modifiedBinding.name);
-        let funcInstance = additionalFunctionInfo.instance;
-        invariant(funcInstance !== undefined);
-        funcInstance.residualFunctionBindings.set(modifiedBinding.name, residualBinding);
-        let newValue = modifiedBinding.value;
-        invariant(newValue);
-        newValue = this.visitEquivalentValue(newValue);
-        residualBinding.modified = true;
-        let otherFunc = residualBinding.additionalFunctionOverridesValue;
-        if (otherFunc !== undefined && otherFunc !== functionValue) {
-          let otherNameVal = otherFunc._SafeGetDataPropertyValue("name");
-          let otherNameStr = otherNameVal instanceof StringValue ? otherNameVal.value : "unknown function";
-          let funcNameVal = functionValue._SafeGetDataPropertyValue("name");
-          let funNameStr = funcNameVal instanceof StringValue ? funcNameVal.value : "unknown function";
-          let error = new CompilerDiagnostic(
-            `Variable ${
-              modifiedBinding.name
-            } written to in optimized function ${funNameStr} conflicts with write in another optimized function ${otherNameStr}`,
-            funcNameVal.expressionLocation,
-            "PP1001",
-            "RecoverableError"
-          );
-          if (functionValue.$Realm.handleError(error) === "Fail") throw new FatalError();
-        }
-        residualBinding.additionalFunctionOverridesValue = functionValue;
-        additionalFunctionInfo.modifiedBindings.set(modifiedBinding, residualBinding);
-        // TODO nested optimized functions: revisit adding GLOBAL as outer optimized function
-        residualBinding.potentialReferentializationScopes.add("GLOBAL");
-        return [residualBinding, newValue];
+        let fixpoint_rerun = () => {
+          if (this.hasBinding(modifiedBinding.environment, modifiedBinding.name)) {
```

Comment:
This is the main change, everything follows from this: Make the body of `visitModifiedBinding` participate in the fixed point computation, and be conditional on whether the binding has been visited for other reasons.